### PR TITLE
Integrate ARKit and use if it's device compatable

### PR DESCRIPTION
**Pull request by callum-code**
This moves the UI in `FacesViewController` over to a common class `FaceViewer` with protocol stubs that are implemented by both `FacesViewController` and `FacesARViewController`. On startup, if ARKit face tracking is supported on device it opens the ARKit one and if not it opens the ARCore one :)

### CODE REVIEW

**Review by mamaro556 — APPROVED** [review]
LGTM.  I see you broke out the FacesVIewController into the ViewController and View separately.  Nice.

File: iOS/MMIWG2S/ViewControllers/FacesARViewController.swift [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+class FacesARViewController: UIViewController {
+    
+    private let faceNodeName = "faceNode"
+    private let faceViewer = ARFaceViewerUI()
+    
+    private var sceneView: ARSCNView = ARSCNView()
+    private var currentFaceAnchor: ARFaceAnchor?
+    
+    private let imageBlack = UIImage(named: "Face.scnassets/arkit_hand_black.png")
+    private let imageRed = UIImage(named: "Face.scnassets/arkit_hand_red.png")
```
Comment (Mehequanna) [inline iOS/MMIWG2S/ViewControllers/FacesARViewController.swift:21]: ❓ These still have the mesh on them. Will you be able to turn that off or edit it out? Or do you need me to do that?

File: iOS/MMIWG2S/ViewControllers/FacesARViewController.swift [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+extension FacesARViewController: ARSCNViewDelegate {
+    
+    func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
+        guard let faceAnchor = anchor as? ARFaceAnchor else { return }
+        currentFaceAnchor = faceAnchor
+        
+        if node.childNodes.isEmpty,
+           let device = sceneView.device,
+           let faceGeometry = ARSCNFaceGeometry(device: device),
+           let material = faceGeometry.firstMaterial {
```
Comment (Mehequanna) [inline iOS/MMIWG2S/ViewControllers/FacesARViewController.swift:135]: This is really unique. Is the if statement checking on the first and then setting lets to be used? Or is each let part of the if check?

File: iOS/MMIWG2S/Views/ARFaceViewerUI.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    func resumeSessionOnceCaptured()
+}
+
+class ARFaceViewerUI: UIView {
+    private let shareButtonBottomPadding: CGFloat = 50
+    private let shareButtonDiameter: CGFloat = 105
+    private let toggleViewWidth: CGFloat = 97
+    private let toggleViewBottomPadding: CGFloat = 75
+    private let largerColorToggleWidth: CGFloat = 54
+    private let smallerColorToggleWidth: CGFloat = 36
```
Comment (Mehequanna) [inline iOS/MMIWG2S/Views/ARFaceViewerUI.swift:24]: ❓ Would these be ok to move into the SceneButton class as static fields that can be called?

**Review by Mehequanna — APPROVED** [review]
Looks awesome, sharing the image works! I know ArCore still has an issue, but we can get to that down the road :)

If we can get the mesh out of the texture we are using, that would be great.

File: iOS/MMIWG2S/ViewControllers/FacesARViewController.swift (hunk 1 above)
Comment (callum-code) [reply iOS/MMIWG2S/ViewControllers/FacesARViewController.swift:21]: I'd be happy to take it out. I added it in for debugging, but doesn't really seem necessary

File: iOS/MMIWG2S/ViewControllers/FacesARViewController.swift (hunk 2 above)
Comment (callum-code) [reply iOS/MMIWG2S/ViewControllers/FacesARViewController.swift:135]: They're all treated as `if let`

File: iOS/MMIWG2S/Views/ARFaceViewerUI.swift (hunk 3 above)
Comment (callum-code) [reply iOS/MMIWG2S/Views/ARFaceViewerUI.swift:24]: Sure thing

**Pull request merged**